contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,860 @@
+        uint256 tokenId
+    );
+
+    event LiquidityAdded(
+        uint256 WETHAmountDesired,
+        uint256 OETHbAmountDesired,
+        uint256 WETHAmountSupplied,
+        uint256 OETHbAmountSupplied,
+        uint256 tokenId,
+        uint256 underlyingAssets
+    );
+
+    /**
+     * @dev Verifies that the caller is the Governor, or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            msg.sender == governor() ||
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Not the Governor or Strategist"
+        );
+        _;
+    }
+
+    /// @notice the constructor
+    /// @param _stratConfig the basic strategy configuration
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _OETHbAddress Address of the Erc20 OETHb Token contract
+    /// @param _swapRouter Address of the Aerodrome Universal Swap Router
+    /// @param _nonfungiblePositionManager Address of position manager to add/remove
+    ///         the liquidity
+    /// @param _clPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _sugarHelper Address of the Aerodrome Sugar helper contract
+    /// @param _lowerBoundingTick Smaller bounding tick of our liquidity position
+    /// @param _upperBoundingTick Larger bounding tick of our liquidity position
+    /// @param _tickClosestToParity Tick that is closer to 1:1 price parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _OETHbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig)
+    {
+        WETH = _wethAddress;
+        OETHb = _OETHbAddress;
+        swapRouter = ISwapRouter(_swapRouter);
+        positionManager = INonfungiblePositionManager(
+            _nonfungiblePositionManager
+        );
+        clPool = ICLPool(_clPool);
+        helper = ISugarHelper(_sugarHelper);
+        sqrtRatioX96Tick0 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_lowerBoundingTick);
+        sqrtRatioX96Tick1 = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_upperBoundingTick);
+        sqrtRatioX96TickClosestToParity = ISugarHelper(_sugarHelper).getSqrtRatioAtTick(_upperBoundingTick);
+
+        lowerTick = _lowerBoundingTick;
+        upperTick = _upperBoundingTick;
+        tickSpacing = 1;
+    }
+
+    /**
+     * @notice initialize function, to set up initial internal state
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     * @param _clGauge Address of the Aerodrome slipstream pool gauge
+     */
+    function initialize(
```

Comment:
A few checks here would be nice. Especially making sure that `_assets.length == 0` and `_pTokens.length == 0`. Another way would be to just get rid of those two params here and just set it to `new address[]()` internally

Reply:
Great point. Fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/27bc5e77bb85891e4dc1961b0a344793d0654d3e